### PR TITLE
[BREAKING] Disallow sha3 suicide

### LINKED ISSUE #4203: Remove suicide/sha3 aliases
Similar to #3003.

I think two years having `selfdestruct` (and `suicide` being deprecated) should have been enough time to stop using it.

### PR DESCRIPTION
**Pull request by axic**
Closes #4203.

### CODE REVIEW

**Review by ekpyron — DISMISSED** [review]
The Changelog seems to have been messed up before this already - not sure whether it makes more sense to fix it here or separately, but otherwise this looks fine.

File: test/libsolidity/syntaxTests/tight_packing_literals.sol [hunk 1]
```diff
@@ -2,9 +2,6 @@ contract C {
     function f() pure public returns (bytes32) {
         return keccak256(1);
     }
-    function g() pure public returns (bytes32) {
-        return sha3(1);
-    }
     function h() pure public returns (bytes32) {
```
Comment (ekpyron) [inline test/libsolidity/syntaxTests/tight_packing_literals.sol:5]: This could be named ``g`` now, but it doesn't really matter. Same below.

**Comment by axic** [comment]
Seems like the DAO code in solc-js uses `sha3` :)

File: test/libsolidity/syntaxTests/tight_packing_literals.sol (hunk 1 above)
Comment (axic) [reply test/libsolidity/syntaxTests/tight_packing_literals.sol:5]: I think I'll leave it this way to reduce the number of changes.

**Comment by axic** [comment]
@chriseth @ekpyron please review again. This now uses the latest solc-js and as such should pass the tests. Also updated the local imported sources.

**Review by chriseth — DISMISSED** [review]

**Review by ekpyron — DISMISSED** [review]

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 2]
```diff
@@ -3760,27 +3724,6 @@ BOOST_AUTO_TEST_CASE(iterated_keccak256_with_bytes)
 	));
 }
 
-BOOST_AUTO_TEST_CASE(sha3_multiple_arguments)
```
Comment (axic) [inline test/libsolidity/SolidityEndToEndTest.cpp:3763]: There is an identical `keccak256_multiple_arguments` test.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 3]
```diff
@@ -2028,23 +2009,6 @@ BOOST_AUTO_TEST_CASE(keccak256)
 	testContractAgainstCpp("a(bytes32)", f, u256(-1));
 }
 
-BOOST_AUTO_TEST_CASE(sha3)
```
Comment (axic) [inline test/libsolidity/SolidityEndToEndTest.cpp:2031]: There is an identical `keccak256` test.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 4]
```diff
@@ -1971,25 +1971,6 @@ BOOST_AUTO_TEST_CASE(log_in_constructor)
 	BOOST_CHECK_EQUAL(m_logs[0].topics[0], h256(u256(2)));
 }
 
-BOOST_AUTO_TEST_CASE(suicide)
```
Comment (axic) [inline test/libsolidity/SolidityEndToEndTest.cpp:1974]: There is an identical `selfdestruct` test.

**Review by ekpyron — APPROVED** [review]
Once tests are passing this should still be fine.

**Comment by axic** [comment]
The single test failure is an RPC issue.

**Pull request merged**